### ISSUE TITLE
ConnectionRefusedError

### ISSUE THREAD

**Issue by Dolidodzik** [apragacz/django-rest-registration#78]
(I posted the same on  [stackoverflow](https://stackoverflow.com/questions/58042296/django-rest-registration-0-5-1-throws-reset-password-verification-url-and-reset).

I am creating REST api, with Django Rest Framework, and first of all I have to give users the way to register. I saw django-rest-registration as easy way to make this, but when I did everything as is said in Quickstart, my terminal throws those errors (I tried simpliest solution, because I am looking for really simple way to register new users):

```
django.core.management.base.SystemCheckError: SystemCheckError: System check identified some issues:

ERRORS:
?: (rest_registration.E001) RESET_PASSWORD_VERIFICATION_URL is not set
?: (rest_registration.E004) VERIFICATION_FROM_EMAIL is not set

System check identified 2 issues (0 silenced).
```
Here's my code:

settings.py:

```
INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',

    'rest_framework',
    'corsheaders',

    'rest_registration',

    'app',
]

REST_REGISTRATION = {
    'REGISTER_VERIFICATION_ENABLED': False,
    'REGISTER_EMAIL_VERIFICATION_ENABLED': False,
    'RESET_PASSWORD_VERIFICATION_ENABLED': False,
}
```
urlpatterns:

```
urlpatterns = [
    path('admin/', admin.site.urls),
    path('', include('app.urls')),
     path('accounts/', include('rest_registration.api.urls')),
]
```

But, when I tried to change REST_REGISTRATION to this:

```
REST_REGISTRATION = {
    'REGISTER_VERIFICATION_URL': 'https://frontend-host/verify-user/',
    'RESET_PASSWORD_VERIFICATION_URL': 'https://frontend-host/reset-password/',
    'REGISTER_EMAIL_VERIFICATION_URL': 'https://frontend-host/verify-email/',
     'VERIFICATION_FROM_EMAIL': 'no-reply@example.com',
  }

```
and I submitted the request, I saw another error:

`ConnectionRefusedError at /accounts/register/ `+ Traceback:

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',

    'rest_framework',
    'corsheaders',

    'rest_registration',

    'app',
]

REST_REGISTRATION = {
    'REGISTER_VERIFICATION_ENABLED': False,
    'REGISTER_EMAIL_VERIFICATION_ENABLED': False,
    'RESET_PASSWORD_VERIFICATION_ENABLED': False,
}
urlpatterns:

urlpatterns = [
    path('admin/', admin.site.urls),
    path('', include('app.urls')),
     path('accounts/', include('rest_registration.api.urls')),
]
But, when I tried to change REST_REGISTRATION to this:

REST_REGISTRATION = {
    'REGISTER_VERIFICATION_URL': 'https://frontend-host/verify-user/',
    'RESET_PASSWORD_VERIFICATION_URL': 'https://frontend-host/reset-password/',
    'REGISTER_EMAIL_VERIFICATION_URL': 'https://frontend-host/verify-email/',

        'VERIFICATION_FROM_EMAIL': 'no-reply@example.com',
    }
and I submitted the request, I saw another error:

ConnectionRefusedError at /accounts/register/ + Traceback:

```
Environment:


Request Method: POST
Request URL: http://localhost:8000/accounts/register/

Django Version: 2.2.3
Python Version: 3.7.3
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'rest_framework',
 'corsheaders',
 'rest_registration',
 'app']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'corsheaders.middleware.CorsMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback:

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/django/views/decorators/csrf.py" in wrapped_view
  54.         return view_func(*args, **kwargs)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/django/views/generic/base.py" in view
  71.             return self.dispatch(request, *args, **kwargs)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/rest_framework/views.py" in dispatch
  495.             response = self.handle_exception(exc)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/rest_framework/views.py" in handle_exception
  455.             self.raise_uncaught_exception(exc)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/rest_framework/views.py" in dispatch
  492.             response = handler(request, *args, **kwargs)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/rest_framework/decorators.py" in handler
  55.             return func(*args, **kwargs)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/rest_registration/api/views/register.py" in register
  96.         send_verification_notification(user, signer, template_config)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/rest_registration/notifications/email.py" in send_verification_notification
  24.     send_notification(notification)

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/rest_registration/notifications/email.py" in send_notification
  209.     notification.send()

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/django/core/mail/message.py" in send
  291.         return self.get_connection(fail_silently).send_messages([self])

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/django/core/mail/backends/smtp.py" in send_messages
  103.             new_conn_created = self.open()

File "/home/fullnamedebian/.local/lib/python3.7/site-packages/django/core/mail/backends/smtp.py" in open
  63.             self.connection = self.connection_class(self.host, self.port, **connection_params)

File "/usr/lib/python3.7/smtplib.py" in __init__
  251.             (code, msg) = self.connect(host, port)

File "/usr/lib/python3.7/smtplib.py" in connect
  336.         self.sock = self._get_socket(host, port, self.timeout)

File "/usr/lib/python3.7/smtplib.py" in _get_socket
  307.                                         self.source_address)

File "/usr/lib/python3.7/socket.py" in create_connection
  727.         raise err

File "/usr/lib/python3.7/socket.py" in create_connection
  716.             sock.connect(sa)

Exception Type: ConnectionRefusedError at /accounts/register/
Exception Value: [Errno 111] Connection refused
```

PS: When I logged to django admin I saw users are created when I submit request on /accounts/register/. 

![image](https://user-images.githubusercontent.com/45403119/65387061-bd90d680-dd32-11e9-9074-2b1a81cc6316.png)

**Comment by apragacz** [comment]
hi @Dolidodzik,
Thank you for your detailed report! I really apprieciate the effort.

I have no idea why the minimal example didn't work (That you got these system check errors; you should get none). Do you remember where you get the config from? I noticed that the config provided in the README is invalid (the config in the docs is OK on the other hand), And it could generate the system check errors you posted above. I will fix that to avoid further confusion.

As for the case with the e-mail:

*   Are you sure you configured your e-mail server correctly in your Django app, like described here?
    https://docs.djangoproject.com/en/dev/topics/email/

*   `no-reply@example.com` is obviously an example e-mail and unless you own example.com domain (I'm pretty sure you don't) you shouldn't use it. If you just want to test the flow on your local development setup, you can use one of the debug email backends like the console backend for instance:
https://docs.djangoproject.com/en/dev/topics/email/#console-backend

So it seems that the e-mail cannot be sent because of the misconfiguration, could you confirm it?

The issue that the e-mail is not sent but the user is created is another one; It should be atomic, i.e. the user should not be created when the e-mail wasn't sent successfully (in case of enabled verification ofc.).

**Comment by apragacz** [comment]
BTW: I fixed the README via commit 415a7c3370ec98fef8b5ef3034411f6849be2bbe.
